wasm/microwasm/operations.go

```diff
@@ -0,0 +1,591 @@
+
+func (o *OperationDrop) Kind() OperationKind {
+	return OperationKindDrop
+}
+
+type OperationSelect struct{}
+
+func (o *OperationSelect) Kind() OperationKind {
+	return OperationKindSelect
+}
+
+type OperationPick uint32
+
+func (o *OperationPick) Kind() OperationKind {
+	return OperationKindPick
+}
+
+type OperationSwap uint32
+
+func (o *OperationSwap) Kind() OperationKind {
+	return OperationKindSwap
+}
+
+type OperationGlobalGet uint32
+
+func (o *OperationGlobalGet) Kind() OperationKind {
+	return OperationKindGlobalGet
+}
+
+type OperationGlobalSet uint32
+
+func (o *OperationGlobalSet) Kind() OperationKind {
+	return OperationKindGlobalSet
+}
+
+type MemoryImmediate struct {
+	Alignment, Offest uint32
+}
+
+type OperationLoad struct {
+	Type SignLessInt
+	Arg  *MemoryImmediate
+}
+
+func (o *OperationLoad) Kind() OperationKind {
+	return OperationKindLoad
+}
+
+type OperationLoad8 struct {
+	Type SignFulInt
+	Arg  *MemoryImmediate
+}
+
+func (o *OperationLoad8) Kind() OperationKind {
+	return OperationKindLoad8
+}
+
+type OperationLoad16 struct {
+	Type SignFulInt
+	Arg  *MemoryImmediate
+}
+
+func (o *OperationLoad16) Kind() OperationKind {
+	return OperationKindLoad16
+}
+
+type OperationLoad32 struct {
+	Signed bool
+	Arg    *MemoryImmediate
+}
+
+func (o *OperationLoad32) Kind() OperationKind {
+	return OperationKindLoad32
+}
+
+type OperationStore struct {
+	Type SignLessInt
+	Arg  *MemoryImmediate
+}
+
+func (o *OperationStore) Kind() OperationKind {
+	return OperationKindStore
+}
+
+type OperationStore8 struct {
+	Type SignLessInt
+	Arg  *MemoryImmediate
+}
+
+func (o *OperationStore8) Kind() OperationKind {
+	return OperationKindStore8
+}
+
+type OperationStore16 struct {
+	Type SignLessInt
+	Arg  *MemoryImmediate
+}
+
+func (o *OperationStore16) Kind() OperationKind {
+	return OperationKindStore16
+}
+
+type OperationStore32 struct {
+	Arg *MemoryImmediate
+}
+
+func (o *OperationStore32) Kind() OperationKind {
+	return OperationKindStore32
+}
+
+type OperationMemorySize struct{}
+
+func (o *OperationMemorySize) Kind() OperationKind {
+	return OperationKindMemorySize
+}
+
+type OperationMemoryGrow struct{}
+
+func (o *OperationMemoryGrow) Kind() OperationKind {
+	return OperationKindMemoryGrow
+}
+
+type OperationConstI32 uint32
+
+func (o *OperationConstI32) Kind() OperationKind {
+	return OperationKindConstI32
+}
+
+type OperationConstI64 uint64
+
+func (o *OperationConstI64) Kind() OperationKind {
+	return OperationKindConstI64
+}
+
+type OperationConstF32 float32
+
+func (o *OperationConstF32) Kind() OperationKind {
+	return OperationKindConstF32
+}
+
+type OperationConstF64 float64
+
+func (o *OperationConstF64) Kind() OperationKind {
+	return OperationKindConstF64
+}
+
+type OperationEq SignLessInt
+
+func (o *OperationEq) Kind() OperationKind {
+	return OperationKindEq
+}
+
+type OperationNe SignLessInt
+
+func (o *OperationNe) Kind() OperationKind {
+	return OperationKindNe
+}
+
+type OperationEqz SignLessInt
```

Comment:
if at the top of the file, there's a link to the overall spec, I don't mind adding a commit to backfill comments per type. They would be like

```
// OperationEqz implements X ex foo = bar
// See https:// direct path to OperationEqz
```

we can assume that unlike GitHub code links spec links are permalinks even if sometimes they mess that up, it is usually mechanical to fix. GitHub code links are less ideal